Assets/Scripts/Behaviours/CohesionBehaviour.cs

```diff
@@ -0,0 +1,25 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+[CreateAssetMenu(menuName = "Flock/Behaviour/Cohesion")]
+public class CohesionBehaviour : FilteredFlockBehaviour
+{
+    public override Vector3 CalculateMove(FlockAgent agent, List<Transform> context, Flock flock)
+    {
+        List<Transform> filteredContext = (filter == null) ? context : filter.Filter(agent, context);
```

Comment:
Was doing a bit of reading into C#. I think the better way of doing null checks is using the keyword `is` , and `is not` for the opposite.